app/src/main/java/com/example/parkingsystemkotlin/database/ParkingDatabase.kt

```diff
@@ -8,13 +8,14 @@ object ParkingDatabase {
     private var parkingLots: Int = Constant.PARKING_LOT_NOT_SET
 
     fun addReservation(reservation: Reservation) {
-        val lot = reservation.parkingLot
-        if (hashMapReservation.containsKey(lot)) {
-            this.hashMapReservation[lot]?.add(reservation)
-        } else {
+        with(reservation.parkingLot) {
             val arraylist = ArrayList<Reservation>()
-            arraylist.add(reservation)
-            this.hashMapReservation[lot] = arraylist
+            if (hashMapReservation.containsKey(this)) {
+                hashMapReservation[this]?.add(reservation)
+            } else {
+                arraylist.add(reservation)
+                hashMapReservation.put(this, arraylist)
```

Comment:
instead of creating a new arraylist here you can set  
hashMapReservation.put(this, listOf(reservation))

Reply:
Done